src/main/java/org/terasology/climateConditions/HumidityGeneratorSystem.java

```diff
@@ -39,12 +45,43 @@
     @Override
     public void preBegin() {
         environmentSystem.addHumidityModifier(1000,
-                new ConditionModifier() {
-                    @Override
-                    public float getCondition(float value, float x, float y, float z) {
-                        return getValue(value, x, y, z);
+            new ConditionModifier() {
+                @Override
+                public float getCondition(float value, float x, float y, float z) {
+                    Vector3i position = new Vector3i(x, y, z);
+                    //Perfectly humid if the block is water
+                    Block currentBlock = environmentSystem.getWorld().getBlock(position);
+                    if (currentBlock != null) {
+                        if (currentBlock.getDisplayName().contains("Water")) {
+                            return 1;
+                        }
+                    }
+                    // Reduce humidity (humidity = relative humidity) when temperature is higher
+                    float modifier = (environmentSystem.temperatureBase - environmentSystem.getTemperature(x,
+                            y, z)) * .05f * (1 + value);
+
+                    if (environmentSystem.getBiome().getBiome(position).isPresent()) {
+                        // Different biomes indicate different humidities
+                        Biome currentBiome = environmentSystem.getBiome().getBiome(position).get();
+                        if (currentBiome.equals(CoreBiome.OCEAN) || currentBiome.equals(CoreBiome.SNOW)) {
+                            return TeraMath.clamp(.9f + modifier, 0, 1);
+                        } else if (currentBiome.equals(CoreBiome.BEACH)) {
+                            return TeraMath.clamp(.8f + modifier, 0, 1);
+                        } else if (currentBiome.equals(CoreBiome.FOREST)) {
+                            return TeraMath.clamp(.7f + modifier, 0, 1);
+                        } else if (currentBiome.equals(CoreBiome.PLAINS)) {
+                            return TeraMath.clamp(.55f + modifier, 0, 1);
+                        } else if (currentBiome.equals(CoreBiome.MOUNTAINS)) {
+                            return TeraMath.clamp(.3f + modifier, 0, 1);
+                        } else if (currentBiome.equals(CoreBiome.DESERT)) {
+                            return TeraMath.clamp(.15f + modifier, 0, 1);
```

Comment:
Is it possible somehow to introduce more biomes from an outside module?
For eg- if we have rainforests they would have an absurd amount of humidity compared to other forests. Also, I have plans of introducing Volcanoes and it would be weird if it gets rainfall due to high humidity
Also shouldn't mountains have high humidity as well for frequent snowfall? This is however based on my assumption that rainfall and snowfall share common logic in Terasology.

Reply:
I think @Qwertygiy was/is working on making biomes more extensible and world providers configurable via prefabs. 

In this case, as `CoreBiome` is an enum you probably want to use a `switch` statement instead of the `if-else`-cascade.